internal/driver/driver.go

```diff
@@ -102,10 +102,17 @@ func (d *Driver) HandleReadCommands(deviceName string, protocols map[string]mode
 		return responses, err
 	}
 
+	credentials, err := GetCredentials(connectionInfo.CredentialsPath)
```

Comment:
My first reaction was to ask whether not it would possible to request the credentials in the driver's `Initialize` method. That said, I seem to recall from your Secrets Everywhere ADR that `SecretProvider.GetSecrets` caches credentials correct? 

Can secrets be updated? If so, does Vault implement any sort of watcher-type functionality like Consul?

Reply:
Yes, secrets can be updated, but currently these secrets would be updated using the `secret` endpoint which then clears the cache so the new ones are retrieved the next time.